joyent/sdc-cloudapi#86 want support for additional "external" NIC tags

arekinath opened the pull request:
Testing done:
 * `make check`
 * built cloudapi image using `make buildimage`, deployed
 * set SAPI metadata `CLOUDAPI_EXTERNAL_TAGS` on the cloudapi service object
 * verified that cloudapi config was updated and restarted
 * verified that `triton network list` now shows networks on that tag as "public"
 * verified that creating a new VM without any `--network` options now includes a network from that tag

We've had this one in production at UQ since at least mid last year, and previous versions of it (not quite the same as this patch) long before that.

arekinath commented:
Note that this depends on https://github.com/joyent/node-triton-netconfig/pull/4 being merged and released in 1.5.0

Until then you'll need to adjust the `package.json` to point at `git+https://github.com/eait-itig/node-triton-netconfig#eait` to test.

danmcd commented:
Shouldn't you update package.json with the new 1.5.0 requirement?

danmcd commented:
Also it's return of the "make check" hell with copyrights.  :-/

danmcd commented:
I will also have to file a TRITON ticket, since this is the last push related to it. The summary for this push will be altered to reflect that Triton ticket.

bahamat reviewed (changes requested):
Need `make check` fixes, and testing notes, and then I'm pretty much good with this change.

bahamat commented:
I guess "we've had it in production" is qualifies for testing notes. So main thing here is just `make check`.

arekinath commented:
Rebased on latest `master`, no changes made (clean rebase)

`make check` definitely passes as-is, and the dependency I modified reads as "1.5.0" -- I have force-pushed the branch more than once, did you somehow look at an older version?

danmcd commented:
We're having Jenkins weirdness on our side.  Pardon any delays.

danmcd commented:
Aha.

Because of the way check-copyright works, we have this scenario:

- Your commit is dated 2020
- The copyright updates to 2021 DO NOT MATCH the commit date, SO copyright fails.

Pick one of two solutions:
1.) Modify your (rebased?) commit to have a git date of 2021

OR

2.) Change all of the "new" copyright dates to 2020.

Your choice.

danmcd commented:
I created an "alexw" branch with a fixed git commit date and Jenkins loved it.

bahamat commented:
OK, I made #103 to supersede this one so that make check will shut up about the copyright date.

danmcd commented:
Subsumed by #103.
